operator/apis/loki/v1/lokistack_types.go

```diff
@@ -630,6 +630,60 @@ type QueryLimitSpec struct {
 	CardinalityLimit int32 `json:"cardinalityLimit,omitempty"`
 }
 
+// BlockedQueryType defines the query limits type for blocked queries.
+//
+// +kubebuilder:validation:Enum=filter;limited;metric
+type BlockedQueryType string
+
+const (
+	// BlockedQueryFilter defines the blocking type for queries with at least one log filter.
+	BlockedQueryFilter = "filter"
+	// BlockedQueryLimited defines the blocking type for queries without a filter or a metric aggregation.
+	BlockedQueryLimited = "limited"
+	// BlockedQueryMetric defines the blocking type for queries with an aggregation.
+	BlockedQueryMetric = "metric"
+)
+
+// BlockedQuerySpec defines the rule spec for queries to be blocked.
+type BlockedQuerySpec struct {
+	// Hash is a 32-bit FNV-1 hash of the query string.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Query Hash"
+	Hash int32 `json:"hash,omitempty"`
+	// Pattern defines the pattern matching the queries to be blocked.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Query Pattern"
+	Pattern string `json:"pattern,omitempty"`
+	// Regex defines if the pattern is a regular expression. If false the pattern will be used only for exact matches.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch",displayName="Regex"
+	Regex bool `json:"regex,omitempty"`
+	// Types defines the list of query types that should be considered for blocking.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Query Types"
+	Types []BlockedQueryType `json:"types,omitempty"`
```

Comment:
Can users provide duplicate types? Does it have any impact on the final config?

Reply:
:thinking: That's a good question, will check this out on the Loki code base. If it is an error state, we should add some validation code.

Reply:
@JoaoBraveCoding Loki does not care if we add the same value multiple times.